From bd6cacdd36615e9b8136aeb86c3924a404785977 Mon Sep 17 00:00:00 2001
From: Hans de Goede <hdegoede@redhat.com>
Date: Tue, 20 Jan 2015 16:18:34 +0100
Subject: startx: Pass -keeptty when telling the server to start on the current
 tty

Detaching from the tty causes systemd-logind to refuse service to the xserver,
the xserver already tries to detect that it is being asked to run on the
current tty and then automatically enables -keeptty, but this code fails if
all of stdin, stdout and stderr are redirected to a file. So explicitly tell
the xserver to not detach when we're telling it to run on the current tty.

BugLink: https://bugzilla.redhat.com/show_bug.cgi?id=1177513
Signed-off-by: Hans de Goede <hdegoede@redhat.com>
Acked-by: Peter Hutterer <peter.hutterer@who-t.net>

diff --git a/startx.cpp b/startx.cpp
index 1c6fce0..45d7bd9 100644
--- a/startx.cpp
+++ b/startx.cpp
@@ -195,7 +195,7 @@ if [ x"$server" = x ]; then
     tty=$(tty)
     if expr match "$tty" '^/dev/tty[0-9]\+$' > /dev/null; then
         tty_num=$(echo "$tty" | grep -oE '[0-9]+$')
-        vtarg="vt$tty_num"
+        vtarg="vt$tty_num -keeptty"
     fi
 #endif
 
-- 
cgit v0.10.2

